process/economics/economicsData.go

```diff
@@ -198,10 +215,16 @@ func (ed *EconomicsData) GasPerDataByte() uint64 {
 
 // ComputeMoveBalanceFee computes the provided transaction's fee
 func (ed *EconomicsData) ComputeMoveBalanceFee(tx process.TransactionWithFeeHandler) *big.Int {
-	gasPrice := big.NewInt(0).SetUint64(tx.GetGasPrice())
-	gasLimit := big.NewInt(0).SetUint64(ed.ComputeGasLimit(tx))
+	return core.SafeMul(tx.GetGasPrice(), ed.ComputeGasLimit(tx))
+}
 
-	return gasPrice.Mul(gasPrice, gasLimit)
+// EstimateMoveBalanceFee estimates the provided transaction's fee
+func (ed *EconomicsData) EstimateMoveBalanceFee(tx process.TransactionWithFeeHandler) *big.Int {
```

Comment:
Wrong function name - we will get into the same problem later on.

Reply:
done